test/integration/helpers/inspec/spec_helper.rb

```diff
@@ -0,0 +1,50 @@
+require 'net/http'
+require 'open-uri'
+require 'uri'
+
+# Copied from jenkins cookbook helper library
+begin
+  open('https://localhost/whoAmI/', ssl_verify_mode: OpenSSL::SSL::VERIFY_NONE)
+rescue SocketError,
+       Errno::ECONNREFUSED,
+       Errno::ECONNRESET,
+       Errno::ENETUNREACH,
+       Errno::EADDRNOTAVAIL,
+       Timeout::Error,
+       OpenURI::HTTPError => e
+  # If authentication has been enabled, the server will return an HTTP
+  # 403. This is "OK", since it means that the server is actually
+  # ready to accept requests.
+  return if e.message =~ /^403/
+
+  puts "Jenkins is not accepting requests - #{e.message}"
+  sleep(0.5)
+  retry
+end
```

Comment:
So here's how I would fix this:
1. Create a new recipe ``jenkins_test::is_ready`` and add this chunk as a ruby file on the system using the ``cookbook_file`` resource. I'd put it somewhere like ``/tmp/is_ready.rb``.
2. Add this recipe to every suite's runlist.
3. Replace this code block with an InSpec ``command`` which does ``/opt/chef/embedded/bin/ruby /tmp/is_ready.rb``. The test should just be that this script exists with a status of 0.

We may want to add a sane loop cutoff in this code so that after X times it errors out.

Reply:
I temporarily added the wanted `cookbook_file` resource under the `default` recipe to test out my attempt first. 6c89e3e

Reply:
Please remove sudo as it's not needed.